sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -884,26 +895,47 @@ private async Task RunAsync(CancellationToken cancellationToken)
                     {
                         if (!ActivePartitionProcessors.TryGetValue(kvp.Key, out var t) || t.Item1.IsCompleted)
                         {
-                            await StopPartitionProcessingIfRunningAsync(kvp.Key, ProcessingStoppedReason.Shutdown).ConfigureAwait(false);
-                            await StartPartitionProcessingAsync(kvp.Key).ConfigureAwait(false);
+                            await StopPartitionProcessingIfRunningAsync(kvp.Key, ProcessingStoppedReason.Shutdown, cancellationToken).ConfigureAwait(false);
+                            await StartPartitionProcessingAsync(kvp.Key, cancellationToken).ConfigureAwait(false);
                         }
                     }))
                     .ConfigureAwait(false);
 
+                cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+
                 // Get a complete list of the partition ids present in the Event Hub.  This should be immutable for the time being, but
                 // it may change in the future.
 
-                var partitionIds = await consumer.GetPartitionIdsAsync().ConfigureAwait(false);
+                var partitionIds = default(string[]);
+
+                try
+                {
+                    partitionIds = await consumer.GetPartitionIdsAsync().ConfigureAwait(false);
```

Comment:
Not applying the `RetryPolicy` here because the `EventHubConsumerClient` already uses it internally.

Reply:
/agree.   Any time that you're delegating to the producer/consumer, you can assume the retries have been applied.